### ISSUE TITLE
`purchaseUpdatedListener()` is returning undefined values

### ISSUE THREAD

**Issue by Mkolakoglu** [dooboolab/react-native-iap#1963]
**Description**
Hi,

I am trying to do a purchase with `RNIap.requestSubscription` and listening the result with `purchaseUpdatedListener()`. But it's returning an object with undefined values for Android:
```
originalTransactionDateIOS: "undefined"
originalTransactionIdentifierIOS: "undefined"
productId: undefined
purchaseToken: ""
quantityIOS: undefined
transactionDate: undefined
transactionId: "undefined"
transactionReceipt: ""
```

However, the subscription is getting complete successfully when I check it from Play Store.

**Expected Behavior**

`purchaseUpdatedListener()` should return meaningful data

**Screenshots**

![Screen Shot 2022-09-15 at 15 31 51](https://user-images.githubusercontent.com/16003411/190418043-4494328f-8ac4-4fa5-9138-f1790456a63f.png)

**Environment:**

- react-native-iap: `11.0.0-rc.1`
- react-native: `0.69.3`
- Platforms (iOS, Android, emulator, simulator, device): `Android`

**To Reproduce**

1. Request the subscription:
```
    const subscriptionOffers = [{
      sku: productId,
      offerToken
    }]

    const result = yield * call(RNIap.requestSubscription, { subscriptionOffers })
 ```
2. Listen the subscription result with:
```
 RNIap.purchaseUpdatedListener((data) => {
    console.log('Result: ', data)
  })
```
3. You will see that it returns undefined values.

**Comment by nes123** [comment]
Same problem on IOS

**Comment by andresesfm** [comment]
@Mkolakoglu can you please share if android-specific fields show the same issue?:
```
  //Android
  productIds?: string[];
  dataAndroid?: string;
  signatureAndroid?: string;
  autoRenewingAndroid?: boolean;
  purchaseStateAndroid?: PurchaseStateAndroid;
  isAcknowledgedAndroid?: boolean;
  packageNameAndroid?: string;
  developerPayloadAndroid?: string;
  obfuscatedAccountIdAndroid?: string;
  obfuscatedProfileIdAndroid?: string;
```

**Comment by andresesfm** [comment]
I have found a possible issue related to this. I have fixed and published version 11.0.0-rc.2` please take a look and see if it fixes it for you

**Comment by Mkolakoglu** [comment]
Hi @andresesfm, thanks for the quick action! I can confirm that it's fixed for Android 🎊
